ui/GeoCounter.cs

```diff
@@ -0,0 +1,116 @@
+using Godot;
+using System;
+
+public partial class GeoCounter : Control
+{
+	[Export]
+	public AnimatedSprite2D GeoIcon;
+	[Export]
+	public Label PlusSignLabel;
+	[Export]
+	public Label CurrentGeoLabel;
+	[Export]
+	public Label GeoToAddLabel;
+
+	private bool IsGeoTransferring = false;
+	private int GeoToAdd = 0;
+	private int CurrentGeo = 0;
+	private int GeoToAddChunkSize = 0;
+
+	// Called every frame. 'delta' is the elapsed time since the previous frame.
+	public override void _Process(double delta)
+	{
+		if (IsGeoTransferring)
+		{
+			int geoToTransfer = (int)Math.Round(GeoToAddChunkSize*(delta/1.5));
+			if (geoToTransfer > 0) // There is enough geo stocked up to transfer a nonzero amount per frame
+			{
+				IsGeoTransferring = TransferGeo(geoToTransfer);
+			}
+			else 
+			{
+				IsGeoTransferring = TransferGeo(1);
+			}
```

Comment:
For handling an edge case like this, since you are calling `TransferGeo()` no matter what, I think it is a little cleaner and easier to read if you update `geoToTransfer` instead of calling the `TransferGeo()` function with a one-off value.

For example: 
```
int geoToTransfer = (int)Math.Round(GeoToAddChunkSize*(delta/1.5));
  
if(geoToTransfer <= 0)
{
  geoToTransfer = 1;
}

TranferGeo(geoToTransfer);
```